### PR TITLE
Make com interfaces ide friendly

### PR DESCRIPTION
**Pull request by feerrenrut**
### Link to issue number:
None
Related: https://github.com/nvaccess/nvda/issues/10916

### Summary of the issue:
When working on modules that make heavy use of `comInterfaces` support from tools or IDEs would be helpful.
The files in `comInterfaces` are generated by `comtypes.gen`.
There are pairs of files, a module named with the `comInterface` ID, and one with a friendly name that imports symbols from the ID file.
The friendly name files generated by `comtypes.gen` expose symbols using runtime logic which some tools can not follow.
Making these symbols appear as errors/warnings about unknown symbols.

### Description of how this pull request fixes the issue:
This PR modifies the build scripts to edit the friendly name file to include a fallback import statement which tools are more likely to be able to follow.

Essentially, the following:
```python
from comtypes.gen import _C523F390_9C83_11D3_9094_00104BD0D535_0_3_0
globals().update(_C523F390_9C83_11D3_9094_00104BD0D535_0_3_0.__dict__)
__name__ = 'comtypes.gen.AcrobatAccessLib'
```
becomes:
```python
try:
	from comtypes.gen import _C523F390_9C83_11D3_9094_00104BD0D535_0_3_0
except ModuleNotFoundError:
	import _C523F390_9C83_11D3_9094_00104BD0D535_0_3_0
	from _C523F390_9C83_11D3_9094_00104BD0D535_0_3_0 import *
globals().update(_C523F390_9C83_11D3_9094_00104BD0D535_0_3_0.__dict__)
__name__ = 'comtypes.gen.AcrobatAccessLib'
```

In addition, I noticed that Accessibility.py is generated, despite being commented out by the sconscript.
Presumably some other interface has a dependence on it?
Regardless this is used heavily in NVDA so it is now explicitly included.

The UIAutomationClient interface is not being generated, and instead was manually checked in.
This is because the version provided by Appveyor (and thus included in NVDA official builds) may not be as up to date as it could be. Instead, as new features of UIA are required this file must be manually updated by generating it on a machine that has the latest version available.

### Testing strategy:
Built and ran locally.
Will test the appveyor build.
Ask community to test building and running on different versions of Windows.

### Known issues with pull request:
None.

### Change log entry:

Changes for devlopers:
```
- comInterfaces are now more easily consumable by developer tools such as IDEs. 
```
### Code Review Checklist:

This checklist is a reminder of things commonly forgotten in a new PR.
Authors, please do a self-review and confirm you have considered the following items.
Mark items you have considered by checking them.
You can do this when editing the Pull request description with an x: `[ ]` becomes `[x]`.
You can also check the checkboxes after the PR is created.

- [x] Pull Request description is up to date.
- [x] Unit tests.
- [x] System (end to end) tests.
- [x] Manual tests.
- [x] User Documentation.
- [x] Change log entry.
- [x] Context sensitive help for GUI changes.

### CODE REVIEW

**Comment by leonardder** [comment]
> The UIAutomationClient interface was not being generated, and instead was manually checked in.

This was and is on purpose. The version of the com interface is tied to the operating system. If a newer version of the interface is released with a new version of Windows, it is usually not yet available in the appveyor image.

**Comment by josephsl** [comment]
Hi, for example, dialog detection through UIA was possible by using UIA COM interfaces from Windows SDK 10.0.17763, and I remember having to update UIA COM interface file then (NVDA 2018.3). This becomes more important once Microsoft releases IUIAutomation7 interface (saw this interface in a WIP build in 2019 but didn’t see it since then), and I look closely at changes to UIAutomationCore.dll whenever dev channel builds come out to spot and support UIA interface changes. When we do update the UIA COM interface, we wait until a Windows 10 feature update with COM interface changes to enter release preview channel before sending in a pull request to update the interface (same can be said about Windows 10 feature update/build map), or in some cases, I do my best to detect parts of the changed COM interface early in WinTenApps add-on. Thanks.

**Comment by lukaszgo1** [comment]
Can't we use .idl files from Windows 10 SDK to generate UIAutomationClient COM interface similar to what were doing for IA2? We can require specific version of Windows 10 SDK and it isn't tied to the version of Windows in use.

**Comment by CyrilleB79** [comment]
Could this PR change the behaviour described in #10833 (scons --clean deletes some files that are fetched from the repo)?
Thanks.

**Comment by michaelDCurran** [comment]
@feerrenrut  yes several other typelibraries would have depended on the accessibility typelibrary. IAccessible2, and UI Automation being two of them.

File: source/comInterfaces_sconscript [hunk 1]
```diff
@@ -26,18 +30,62 @@ def new_my_import(fullname):
...
+	globals().update(_C523F390_9C83_11D3_9094_00104BD0D535_0_3_0.__dict__)
+	__name__ = 'comtypes.gen.AcrobatAccessLib'
+	```
+	Become:
+	``` python
+	try:
+		from comtypes.gen import _C523F390_9C83_11D3_9094_00104BD0D535_0_3_0
+	except ModuleNotFoundError:
+		from _C523F390_9C83_11D3_9094_00104BD0D535_0_3_0 import *
+	globals().update(_C523F390_9C83_11D3_9094_00104BD0D535_0_3_0.__dict__)
```
Comment (michaelDCurran) [inline source/comInterfaces_sconscript:49]: This example is missing an import in the except block, and therefore does not reflect what it really does below.

File: source/comInterfaces_sconscript [hunk 2]
```diff
@@ -26,18 +30,62 @@ def new_my_import(fullname):
...
+	importTemplate = (
+"""try:
+	from comtypes.gen import {libIdentifier}
+except ModuleNotFoundError:
+	import {libIdentifier}
+	from {libIdentifier} import *
+""")
+
+	importPattern = re.compile(r"from comtypes\.gen import ([\w]+)\n")
+	for line in fileinput.input(path, inplace=True):
```
Comment (michaelDCurran) [inline source/comInterfaces_sconscript:63]: A comment here mentioning that fileinput.Input temporarily replaces sys.stdout would be useful.

**Review by michaelDCurran — CHANGES_REQUESTED** [review]
We don't quite know what version of UIAutomationCore Appveyor has... I think its Windows server 2019 or something. Thus we cannot guarantee that UI Automation client will contain all possible interfaces and constants we need.
We need to either keep our pre-generated version of UIAutomationClient COM interfaces, or perhaps include uiAutomationCore.dll in miscDeps... though I think the latter would be breaking licensing restrictions.

File: source/comInterfaces_sconscript (hunk 1 above)
Comment (feerrenrut) [reply source/comInterfaces_sconscript:49]: Yep. I think I can remove this example, I don't think it really helps to clarify anything now.

File: source/comInterfaces_sconscript (hunk 2 above)
Comment (feerrenrut) [reply source/comInterfaces_sconscript:63]: Ah yes, you are right. It is an unintuitive library.

**Review by michaelDCurran — APPROVED** [review]

**Pull request merged**